src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1219,215 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
+                            var tlog = t as ILogSource;
+                            tlog.Initialize();
+                            var logEntries = t.LogEntries;
+                            var jobHistories = AgentUtilities.ConvertToAgentNotebookHistoryInfo(logEntries, job, steps);
+                            Dictionary<string, AgentNotebookHistoryInfo> allNotebookHistoriesHashTable = new Dictionary<string, AgentNotebookHistoryInfo>();
+                            // Finally add the job histories
+                            DataTable materializedNotebookTable = NotebookHistoryDataset.Tables[0];
+                            foreach (DataRow materializedNotebookRow in materializedNotebookTable.Rows){
+                                string materializedRunDateTime = materializedNotebookRow["run_date"].ToString() + materializedNotebookRow["run_time"].ToString();
+                                AgentNotebookHistoryInfo notebookHistory = new AgentNotebookHistoryInfo();
+                                notebookHistory.MaterializedNotebookId = (int)materializedNotebookRow["materialized_id"];
+                                allNotebookHistoriesHashTable.Add(materializedRunDateTime, notebookHistory);
+                            }
+                            foreach (var jobHistory in jobHistories){
+                                string jobRuntime = jobHistory.RunDate.ToString("yyyyMMddHHmmss");
+                                AgentNotebookHistoryInfo notebookHistory = jobHistory;
+                                if(allNotebookHistoriesHashTable.ContainsKey(jobRuntime))
+                                    notebookHistory.MaterializedNotebookId = allNotebookHistoriesHashTable[jobRuntime].MaterializedNotebookId;
+                                notebookHistories.Add(notebookHistory);
+                            }
+                            result.Histories = notebookHistories.ToArray();
+                            result.Success = true;
+                            tlog.CloseReader();
+                            await requestContext.SendResult(result);
+                        }
+                    }
+                }
+                catch(Exception e){
+                    await requestContext.SendError(e);
+                }
+            });
+        }
+
+        internal async Task HandleAgentNotebookMaterializedRequest(AgentNotebookMaterializedParams parameters, RequestContext<AgentNotebookMaterializedResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                try
+                {
+                    var result = new AgentNotebookMaterializedResult();
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                        parameters.OwnerUri,
+                        out connInfo);
+                    if (connInfo != null)
+                    {
```

Comment:
If connInfo is null you never send a response and thus will time out on the ADS side. You should send an error in that case.